src/nupic/bindings/sparse_matrix.i

```diff
@@ -1002,6 +1002,27 @@ def __div__(self, other):
   }
 
 
+  %pythoncode %{
+    def nNonZerosPerRowOnCols(self, rows, cols):
+      rows = numpy.asarray(rows, dtype="uint32")
```

Comment:
I think it is better to have the C++ Numpy classes handle dtype checking or conversion. There is already logic in `NumpyVectorWeakRefT` to make an assertion that the dtype is correct and the `NumpyVector` class will actually convert to the specified dtype if you want that behavior.

And `NumpyVector` will actually convert to the specified dtype so you could use that if you want that behavior.